File: AdvantageFramework/Database/Entities/POP3EmailListenerAccount.vb
Namespace Database.Entities

    <Table("AGY_POP3_ACCOUNT")>
    Public Class POP3EmailListenerAccount
        Inherits BaseClasses.Entity

#Region " Constants "



#End Region

#Region " Enum "

        Public Enum Properties
            ID
            Description
            UserName
            Password
            ReplyTo
            DeleteProcessed
        End Enum

#End Region

#Region " Variables "



#End Region

#Region " Properties "

        <Key>
        <System.ComponentModel.DataAnnotations.Schema.DatabaseGeneratedAttribute(ComponentModel.DataAnnotations.Schema.DatabaseGeneratedOption.Identity)>
        <Required>
        <Column("POP3_ACCOUNT_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(ShowColumnInGrid:=False)>
        Public Property ID() As Integer
        <Required>
        <Column("POP3_ACCOUNT_TYPE")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(ShowColumnInGrid:=False)>
        Public Property AccountType() As Short
        <Required>
        <MaxLength(30)>
        <Column("DESCRIPTION", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsReadOnlyColumn:=True)>
        Public Property Description() As String
        <MaxLength(100)>
        <Column("USERNAME", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity()>
        Public Property UserName() As String
        <Column("PASSWORD", TypeName:="varchar(MAX)")>
        <AdvantageFramework.BaseClasses.Attributes.Entity()>
        Public Property Password() As String
        <MaxLength(50)>
        <Column("REPLY_TO", TypeName:="varchar")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(PropertyType:=BaseClasses.Methods.PropertyTypes.Email)>
        Public Property ReplyTo() As String
        <Column("DELETE_PROCESSED")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(DefaultColumnType:=AdvantageFramework.BaseClasses.DefaultColumnTypes.StandardCheckBox, CustomColumnCaption:="Delete Processed E-Mail Messages")>
        Public Property DeleteProcessed() As Nullable(Of Short)

#End Region

#Region " Methods "

        Public Overrides Function ToString() As String

            ToString = Me.ID.ToString

        End Function
        Public Sub ValidateUserName(ByVal IsValid As Boolean)

            'objects
            Dim ErrorText As String = Nothing

            If Not IsValid Then

                ErrorText = "Please enter a unique POP3 E-Mail Listener Account."

            End If

            _ErrorHashtable("UserName") = ErrorText

            _EntityError = ErrorText

        End Sub

#End Region

    End Class

End Namespace

File: AdvantageFramework.Core/Database/Entities/POP3EmailListenerAccount.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdvantageFramework.Core.Database.Entities
{
    [Table("AGY_POP3_ACCOUNT")]
    public partial class POP3EmailListenerAccount
    {
        [Key]
        [Column("POP3_ACCOUNT_ID")]
        public int Pop3AccountId { get; set; }
        [Column("POP3_ACCOUNT_TYPE")]
        public short Pop3AccountType { get; set; }
        [Required]
        [Column("DESCRIPTION")]
        [StringLength(30)]
        public string Description { get; set; }
        [Column("USERNAME")]
        [StringLength(100)]
        public string Username { get; set; }
        [Column("PASSWORD")]
        public string Password { get; set; }
        [Column("REPLY_TO")]
        [StringLength(50)]
        public string ReplyTo { get; set; }
        [Column("DELETE_PROCESSED")]
        public short? DeleteProcessed { get; set; }
    }
}
